packages/cli/configuration/src/generators-yml/addGenerator.ts

```diff
@@ -1,14 +1,68 @@
 import { TaskContext } from "@fern-api/task-context";
+import Docker from "dockerode";
 import {
     DEFAULT_GROUP_GENERATORS_CONFIG_KEY,
     GeneratorName,
     GeneratorsConfigurationSchema,
     updateGeneratorGroup
 } from ".";
 import { GENERATOR_INVOCATIONS } from "./generatorInvocations";
+import { upgradeGeneratorVersion } from "./upgradeGeneratorVersion";
 
-export function upgradeGenerator(): void {
-    // TODO
+function getOrThrowGeneratorName(generatorName: string, context: TaskContext): GeneratorName {
```

Comment:
nit: I think the convention I've seen elsewhere is `getGeneratorNameOrThrow`.

Reply:
Whooooops, yeah this was not a great name choice on my part